sdk/servicebus/Azure.Messaging.ServiceBus/README.md

```diff
@@ -1,104 +1,361 @@
+
+// deadletter the message, thereby preventing the message from being received again without receiving from the dead letter queue.
+await receiver.DeadLetterAsync(receivedMessage);
+
+// receive the dead lettered message with receiver scoped to the dead letter queue.
+ServiceBusReceiver dlqReceiver = client.GetDeadLetterReceiver(queueName);
+ServiceBusReceivedMessage dlqMessage = await dlqReceiver.ReceiveAsync();
+```
+
+### Using the Processor
+```C# Snippet:ServiceBusProcessMessages
+string connectionString = "<connection_string>";
+string queueName = "<queue_name>";
+// since ServiceBusClient implements IAsyncDisposable we create it with "await using"
+await using var client = new ServiceBusClient(connectionString);
+// get the sender
+ServiceBusSender sender = client.GetSender(queueName);
+
+// create a message batch that we can send
+ServiceBusMessageBatch messageBatch = await sender.CreateBatchAsync();
+messageBatch.TryAdd(new ServiceBusMessage(Encoding.UTF8.GetBytes("First")));
+messageBatch.TryAdd(new ServiceBusMessage(Encoding.UTF8.GetBytes("Second")));
+
+// send the message batch
+await sender.SendBatchAsync(messageBatch);
+
+// get the options to use for configuring the processor
+var options = new ServiceBusProcessorOptions
+{
+    // By default after the message handler returns, the processor will complete the message
+    // If I want more fine-grained control over settlement, I can set this to false.
+    AutoComplete = false,
+
+    // I can also allow for multi-threading
+    MaxConcurrentCalls = 2
+};
+
+// get a processor that we can use to process the messages
+ServiceBusProcessor processor = client.GetProcessor(queueName, options);
+
+// since the message handler will run in a background thread, in order to prevent
+// this sample from terminating immediately, we can use a task completion sources that
+// we complete from within the message handler.
+TaskCompletionSource<bool> tcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+processor.ProcessMessageAsync += MessageHandler;
+processor.ProcessErrorAsync += ErrorHandler;
+
+async Task MessageHandler(ProcessMessageEventArgs args)
+{
+    string body = Encoding.Default.GetString(args.Message.Body.ToArray());
+    Console.WriteLine(body);
+
+    // we can evaluate application logic and use that to determine how to settle the message.
+    await args.CompleteAsync(args.Message);
+    tcs.SetResult(true);
+}
+
+Task ErrorHandler(ProcessErrorEventArgs args)
+{
+    // the error source tells me at what point in the processing an error occurred
+    Console.WriteLine(args.ErrorSource);
+    Console.WriteLine(args.Exception.ToString());
+    return Task.CompletedTask;
+}
+await processor.StartProcessingAsync();
+// await our task completion source task so that the message handler will be invoked at least once.
+await tcs.Task;
+// stop processing once the task completion source was completed.
+await processor.StopProcessingAsync();
 ```
```

Comment:
I think we can provide one example of Azure Active Directory authentication support. What do you think?